packages/block-editor/src/store/selectors.js

```diff
@@ -114,19 +115,76 @@ export function isBlockValid( state, clientId ) {
  * Returns a block's attributes given its client ID, or null if no block exists with
  * the client ID.
  *
+ * Process block bindings to modify the value of the attributes if needed.
+ *
  * @param {Object} state    Editor state.
  * @param {string} clientId Block client ID.
  *
  * @return {Object?} Block attributes.
  */
-export function getBlockAttributes( state, clientId ) {
-	const block = state.blocks.byClientId.get( clientId );
-	if ( ! block ) {
-		return null;
-	}
+export const getBlockAttributes = createRegistrySelector(
+	( select ) => ( state, clientId ) => {
+		// TODO: Check how to properly access the block context.
+		const blockContext = {};
+		const block = state.blocks.byClientId.get( clientId );
+		if ( ! block ) {
+			return null;
+		}
 
-	return state.blocks.attributes.get( clientId );
-}
+		const blockAttributes = state.blocks.attributes.get( clientId );
+
+		// Change attribute values if bindings are present.
+		const bindings = blockAttributes?.metadata?.bindings;
+		if ( ! bindings ) {
+			return blockAttributes;
+		}
+
+		const sources = unlock(
+			select( blocksStore )
+		).getAllBlockBindingsSources();
+
+		for ( const [ attributeName, boundAttribute ] of Object.entries(
+			bindings
+		) ) {
+			const source = sources[ boundAttribute.source ];
+			if (
+				! source?.getValue ||
+				! canBindAttribute( block.name, attributeName )
+			) {
+				continue;
+			}
+
+			const context = {};
```

Comment:
This is still a to do, right?

Reply:
Yes. I added a TODO comment at [this point](https://github.com/WordPress/gutenberg/pull/61817/files/0e91129ee1b9fa7cf34db6567b9dd70e1d6599c6#diff-aa92c48095c35f17feb834dd919baf52274a73c7b1180ff5acce7b6087b126cfR127-R128).